clients/common/cblas_interface.cpp

```diff
@@ -724,6 +724,70 @@ void cblas_gemm<rocblas_double_complex>(rocblas_operation transA,
                 ldc);
 }
 
+template <>
+void cblas_gemm<int8_t, int32_t>(rocblas_operation transA,
+                                 rocblas_operation transB,
+                                 rocblas_int m,
+                                 rocblas_int n,
+                                 rocblas_int k,
+                                 int32_t alpha,
+                                 int8_t* A,
+                                 rocblas_int lda,
+                                 int8_t* B,
+                                 rocblas_int ldb,
+                                 int32_t beta,
+                                 int32_t* C,
+                                 rocblas_int ldc)
+{
+    // cblas does not support int8_t input / int32_t output, however non-overflowing
+    // 32-bit integer operations can be represented accurately with double-precision
+    // floats, so convert to doubles and downcast result down to int32_t.
+    // NOTE: This will not properly account for 32-bit integer overflow, however
+    //       the result should be acceptable for testing.
+
+    int const sizeA = ((transA == rocblas_operation_none) ? k * lda : m * lda);
+    int const sizeB = ((transB == rocblas_operation_none) ? n * ldb : k * ldb);
+    int const sizeC = n * ldc;
+
+    std::unique_ptr<double[]> A_double(new double[sizeA]());
+    std::unique_ptr<double[]> B_double(new double[sizeB]());
+    std::unique_ptr<double[]> C_double(new double[sizeC]());
+
+    for(int i = 0; i < sizeA; i++)
```

Comment:
I think this will only work if transA == T. If transA == N you need tripple loop, one for m, two for k. The two for k account for a unit stride between 4 consecutive values, and a lda stride between non-consecutive values, something like

for(i = 1; i < m; i++){
for(j1 = 0; j1 < 4; j1++){
for(j2 = 0; j2 < k; j2+=4){
A_double[i+(j2+j1)*lda) = static_cast<double>(A[i+j1+j2*lda]);
}}}

Please check the above, I am writing it down quickly.

Reply:
The multiplication asterisk before lda in above equation seems to have been lost in formatting, it should be

i+(j2+j1) multiplied by lda

Reply:
also missing   j2 multiplied by lda